src/service_mongoose_system_stats.erl

```diff
@@ -0,0 +1,209 @@
+-module(service_mongoose_system_stats).
+-author('jan.ciesla@erlang-solutions.com').
+
+-behaviour(gen_server).
+
+-define(BASE_URL, "https://www.google-analytics.com/batch").
+-define(TRACKING_ID, "UA-151671255-1").
+-define(DEFAULT_REPORT_AFTER, 60 * 60 * 1000).
+-define(STAT_TYPE, [mongoose_system_stats]).
+
+-include("mongoose.hrl").
+
+-export([start/1, stop/0]).
+-export([start_link/0, init/1, handle_event/4, handle_continue/2, handle_call/3, handle_cast/2, handle_info/2, terminate/2]).
+
+-record(state, {
+    client_id = '',
+    reports = [],
+    report_after = ?DEFAULT_REPORT_AFTER,
+    loop_timer_ref
+    }).
+-record(service_mongoose_system_stats, {key, value}).
+
+-spec start([]) -> {ok, pid()}.
+start([]) ->
+    Spec = {?MODULE, {?MODULE, start_link, []}, temporary, brutal_kill, worker, [?MODULE]},
+    {ok, _} = ejabberd_sup:start_child(Spec).
+
+stop() ->
+    ejabberd_sup:stop_child(?MODULE).
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+init(_Args) ->
+    {ok, no_state , {continue, do_init}}.
+
+handle_continue(do_init, NoState ) ->
```

Comment:
mongoose_service expects ok to be returned after the function starts. Init might not be the best place for doing telemetry attach and getting client ID, just in case something goes wrong. Handle continue might not be the most popular in gen_server but don't see anything wrong with using it.

Reply:
added empty state to return of init